python/cudf/cudf/core/dataframe.py

```diff
@@ -734,6 +734,10 @@ def __init__(
                 rangeindex = isinstance(
                     columns, (range, pd.RangeIndex, cudf.RangeIndex)
                 )
+                if len(columns) == 0:
+                    empty_dtype = getattr(columns, "dtype", None)
+                else:
+                    empty_dtype = None
```

Comment:
If the columns are non-empty and have a dtype, should we respect that? It feels like there are still times in which we might lose the dtype, or do I have it wrong?

Reply:
Good point. Currently a column passed in as `int8` would be returned as `int64`. I'll push an update to avoid this upcasting